crates/zizmor/src/main.rs

```diff
@@ -141,6 +143,22 @@ struct App {
     /// to audit the repository at a particular git reference state.
     #[arg(required = true)]
     inputs: Vec<String>,
+
+    /// Apply fixes automatically if available.
+    #[arg(long)]
+    fix: bool,
+
+    /// Skip confirmation prompt when applying fixes.
+    #[arg(long, requires = "fix")]
+    yes: bool,
+
+    /// Show what fixes would be applied without actually writing them.
+    #[arg(long, requires = "fix")]
+    dry_run: bool,
+
+    /// Show a diff of the changes before applying them.
+    #[arg(long, requires = "fix")]
+    diff: bool,
```

Comment:
Mind deferring these to a future PR, just to keep this one small(er)? I think we can start with `--fix` and then look at working these in.

Reply:
I separated the functionality into a `fix.go` file for easier review. Also removed the extra flags.